src/app/app-components/login/login-page.component.ts

```diff
@@ -9,15 +11,22 @@ import { Router } from '@angular/router';
     styleUrls: ['./login-page.component.scss'],
 })
 export class LoginPageComponent {
-
-    email: string;
-    password: string;
+    readonly computeControlValidity = computeControlValidity;
+    readonly emailControl = new FormControl({ value: null, disabled: false }, Validators.required);
+    readonly passwordControl = new FormControl({ value: null, disabled: false }, Validators.required);
+    readonly formGroup = new FormGroup({
+        email: this.emailControl,
+        password: this.passwordControl,
+    });
     @Output() loggedIn = new EventEmitter<void>();
 
-    constructor(private readonly authService: AuthorizationService, private readonly router: Router) { }
+    constructor(
+        private readonly authService: AuthorizationService,
+        private readonly router: Router
+    ) { }
 
     onLogin() {
-        this.authService.login(this.email, this.password);
+        this.authService.login(this.emailControl.value, this.passwordControl.value);
```

Comment:
Now the `login` method could receive an `object` with `email` and `password` props which we can take from the `formGroup`.

Reply:
That's a great idea - I'll refactor this.